boards/arm/sama5/giant-board/src/sam_bringup.c

```diff
@@ -0,0 +1,401 @@
+      if (ret != OK)
+        {
+          syslog(LOG_ERR,
+                 "ERROR: Failed to bind SDIO to the MMC/SD driver (slot 0): "
+                 "%d\n",
+                 ret);
+        }
+    }
+
+#ifdef CONFIG_SAMA5D27_SDMMC1_MOUNT
+  /* Mount the volume on SDMMC1 */
+
+  ret = mount(CONFIG_SAMA5D27_SDMMC1_MOUNT_BLKDEV,
+              CONFIG_SAMA5D27_SDMMC1_MOUNT_MOUNTPOINT,
+              CONFIG_SAMA5D27_SDMMC1_MOUNT_FSTYPE,
+              0, NULL);
+
+  if (ret < 0)
+    {
+      _err("ERROR: Failed to mount %s: %d\n",
+           CONFIG_SAMA5D27_SDMMC1_MOUNT_MOUNTPOINT, errno);
+    }
+#endif
+#endif
+
+  return OK;
+}
+#else
+#  define nsh_sdmmc_initialize() (OK)
+#endif
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: sam_bringup
+ *
+ * Description:
+ *   Bring up board features
+ *
+ ****************************************************************************/
+
+int sam_bringup(void)
+{
+  int ret;
+
+  /* Register I2C drivers on behalf of the I2C tool */
+
+  sam_i2ctool();
+
+#ifdef HAVE_SDMMC
+#ifdef CONFIG_SAMA5_SDMMC
+  /* Initialize SDMCC-based MMC/SD card support */
+
+  nsh_sdmmc_initialize();
+#endif
+#endif
+
+#ifdef HAVE_AUTOMOUNTER
+  /* Initialize the auto-mounter */
+
+  sam_automount_initialize();
+#endif
+
+#ifdef HAVE_ROMFS
+  /* Create a ROM disk for the /etc filesystem */
+
+  ret = romdisk_register(CONFIG_GIANT_BOARD_ROMFS_ROMDISK_MINOR, romfs_img,
+                         NSECTORS(romfs_img_len),
+                         CONFIG_GIANT_BOARD_ROMFS_ROMDISK_SECTSIZE);
+  if (ret < 0)
+    {
+      _err("ERROR: romdisk_register failed: %d\n", -ret);
+    }
+  else
+    {
+      /* Mount the file system */
+
+      ret = mount(CONFIG_GIANT_BOARD_ROMFS_ROMDISK_DEVNAME,
+                  CONFIG_GIANT_BOARD_ROMFS_MOUNT_MOUNTPOINT,
+                  "romfs", MS_RDONLY, NULL);
+      if (ret < 0)
+        {
+          _err("ERROR: mount(%s,%s,romfs) failed: %d\n",
+               CONFIG_GIANT_BOARD_ROMFS_ROMDISK_DEVNAME,
+               CONFIG_GIANT_BOARD_ROMFS_MOUNT_MOUNTPOINT, errno);
+        }
+    }
+#endif
+
+#ifdef HAVE_USBHOST
+  /* Initialize USB host operation.  sam_usbhost_initialize() starts a thread
+   * will monitor for USB connection and disconnection events.
+   */
+
+  ret = sam_usbhost_initialize();
+  if (ret != OK)
+    {
+        _err("ERROR: Failed to initialize USB host: %d\n", ret);
+    }
+#endif
+
+#ifdef HAVE_USBMONITOR
+  /* Start the USB Monitor */
+
+  ret = usbmonitor_start();
+  if (ret != OK)
+    {
+        _err("ERROR: Failed to start the USB monitor: %d\n", ret);
+    }
+#endif
+
+#ifdef HAVE_MAXTOUCH
+  /* Initialize the touchscreen */
+
+  ret = sam_tsc_setup(0);
+  if (ret < 0)
+    {
+      syslog(LOG_ERR, "ERROR: sam_tsc_setup failed: %d\n", ret);
```

Comment:
```suggestion
      tmrerr(LOG_ERR, "ERROR: sam_tsc_setup failed: %d\n", ret);
```

Reply:
@patacongo will fix. (This came from SAMA5D2-XULT.)